models/items.js

```diff
@@ -14,24 +12,26 @@ const addTagsToItem = async (itemData) => {
   try {
     const { itemId, itemType, tagPayload } = itemData;
     const batch = firestore.batch();
+    let isNewtag = false;
     for (const tag of tagPayload) {
-      const tagData = await tagModel.doc(tag.tagId).get();
-      const levelData = await levelModel.doc(tag.levelId).get();
+      const itemData = await itemTagsModel
+        .where("itemId", "==", itemId)
+        .where("tagId", "==", tag.tagId)
+        .where("levelId", "==", tag.levelId)
+        .get();
+      if (!itemData.empty) continue;
+      isNewtag = true;
```

Comment:
if the data is not empty then we are marking the new tag as true? shouldn't it be the opposite if the item data of the specified tag is empty i.e. not exists then it is a new tag?

Reply:
Sorry didn't get you. If the `if` statement is executed, the `isNewTag` value never changes. It changes only when `if` isn't true i.e. a new tag. Hence, `isNewTag = true`